### ISSUE TITLE
Android - facebook login dialog misbehaves and results in an error response if the dialog is launched in a different screen orientation than the app's screen orientation.

### ISSUE THREAD

**Issue by msesen** [defold/extension-facebook#63]
- Game is launched in landscape mode.
- Screen rotation is locked on android device.
- Upon clicking on the “Facebok login” button, the login dialog is launched in portrait mode.
- If already logged in, Facebook informs you that you have previously logged into this app. Clicking on the Continue button closes the dialog and immidiately launches the dialog again. Clicking on the Continue button for the second time closes the dialog with an error {status = 0, error = “Aborted”}.

If I disable the screen rotration lock, then it works (if the device is held in landscape mode, otherwise the dialog will launch in portrait mode). The login dialog is launched in landscape mode, clicking on the Continue button returns to the app with the success payload of status = 200. HTML 5 build also works. But unfortunately, I need the game in landscape mode.

Could it be that the Activity that launches the dialog gets destroyed upon screen rotation?

Defold Editor 1.9.0, and the extension version is 9.0.0.

Here is a sample code.

`local function facebookLogin(self, data)

    if data.status == facebook.STATE_OPEN then
        -- Logged in ok.

    elseif data.status == facebook.STATE_CLOSED_LOGIN_FAILED then
        -- Do something to indicate that login failed

    end

    if data.error then
        -- .. something went wrong
    end
end

local function onFacebookLoginBtnClick(self)
    if not facebook then
        -- Facebook login is not available for this device.
        return
    end

    -- Login with read permissions.
    local permissions = { "public_profile" }
    facebook.login_with_permissions(permissions, facebook.AUDIENCE_EVERYONE, facebookLogin)
end`

**Comment by msesen** [comment]
Any update/progress on this by any chance?

**Comment by msesen** [comment]
This is the log I get when auto rotate is disabled on the device and the game is set to run in landcape mode only. Facebook login modal launches in portrait (the defold app is in lanscape):

W defold  : WARNING:SCRIPT: Failed to invoke callback (it was not registered)
W defold  : WARNING:SCRIPT: Failed to unregister callback (it was not registered)

**Comment by AGulev** [comment]
@britzl maybe you know something about it?

**Comment by britzl** [comment]
> [@britzl](https://github.com/britzl) maybe you know something about it?

No, but I suppose we have to look into it. I do recall issues in the past with Facebook logins when changing activity.

@msesen can you please provide a minimal project where we can observe this behavior?

**Comment by msesen** [comment]
@britzl sure, will do as soon as I get the chance, super busy with life and work at the moment. Until then here is the basic setup:

1. Create a new project with the following:
game.project -> Display -> Width: 960
game.project -> Display-> Height: 540
game.project -> Display-> Dynamic Orientation: NOT ENABLED / UNTICKED

2. Setup a facebook login as per the Facebook extension documentation. Add a GUI button so upon clicking, it will invoke the Facebook login dialog.

3. Export and install the app on your Android device. Make sure the Auto Rotate on the device is disabled -> device always stays in Portrait.

Start the app on the installed device which should be launched in landscape. Click on the GUI button to invoke the Facebook login dialog which should be launched in portrait.

**Comment by msesen** [comment]
[defold_facebook_login.zip](https://github.com/user-attachments/files/30655837/defold_facebook_login.zip)

@britzl here is a minimal project. To test this, you will need to provide the appid and the clienttoken in gameproject.

[facebook]
appid = 
clienttoken =
autoinit = 1 

Still getting the following log from Android

defold  : WARNING:SCRIPT: Failed to invoke callback (it was not registered)
defold  : WARNING:SCRIPT: Failed to unregister callback (it was not registered)
defold  : WARNING:SCRIPT: Failed to invoke callback (it was not registered)
defold  : WARNING:SCRIPT: Failed to unregister callback (it was not registered)

After 3 years I am about to release my game but this is kinda holding me back. Any workaround for this will be appreciated!
